### ISSUE TITLE
Weight normalization layer

### ISSUE THREAD

**Issue by nashory** [nashory/progressive-growing-torch#2]
(https://github.com/torch/nn/blob/master/WeightNorm.lua)

**Comment by nashory** [comment]
normalize the weights of the previous layer dynamically.
need to be implemented.

**Comment by nashory** [comment]
Can I use simply use [torch nn.WeightNorm](https://github.com/torch/nn/blob/master/doc/containers.md#weight-normalization) for equalized learning rate in the paper??

**Comment by nashory** [comment]
tested [torch nn.WeightNorm](https://github.com/torch/nn/blob/master/doc/containers.md#weight-normalization), but it seems to harm training.

**Comment by nashory** [comment]
I found (https://github.com/stormraiser/GAN-weight-norm)
maybe we can use this.
